dotnet/src/Functions/Functions.UnitTests/Markdown/Functions/KernelFunctionMarkdownTests.cs

```diff
@@ -47,7 +100,11 @@ These are AI execution settings
         {
             "service1" : {
                 "model_id": "gpt4",
-                "temperature": 0.7
+                "temperature": 0.7,
+                "function_choice_behavior": {
+                    "type": "auto",
+                    "functions": ["p1.f1"]
```

Comment:
Is there a test where the `functions` property is an empty array and one where it isn't provided?

Reply:
No, there were none. I've added the requested tests for each behavior.